docs/source/catalog.rst

```diff
@@ -136,11 +131,88 @@ returned data.  Each data source has several attributes:
 
 - ``parameters``: A dictionary of data source parameters.  See below for more details.
 
-Parameters allow the user to customize the data returned by a data source.  Most often, parameters are used to filter
-or reduce the data in specific ways defined by the catalog author.  The parameters defined for a given data source are
-available for use in template strings, which can be used to alter the arguments provided to the driver.  For example,
-a data source might accept a "postal_code" argument which is used to alter a database query, or select a particular
-group within a file.  Users set parameters with keyword arguments to the ``get()`` method on the catalog object.
+
+Templating
+''''''''''
+
+Intake catalog files support Jinja2 templating for driver arguments. Any occurrence of
+a substring like ``{{field}}`` will be replaced by the value of the user parameters with
+that same name, or explicitly provided by the user. For how to specify these user parameters,
+see the next section.
+
+Some additional values are available for templating. The following is always available:
+
+- ``CATALOG_DIR``: The full path to the directory containing the YAML catalog file.  This is especially useful
+  for constructing paths relative to the catalog directory to locate data files and custom drivers.
+
+so that the search for CSV files for the two "entry1" blocks, above, will happen in the same directory as
+where the catalog file was found.
+
+The following functions `may` be available. Since these execute code, the user of a catalog may decide
+whether they trust those functions or not.
+
+- ``env("USER")``: look in the environment for the named variable
+- ``client_env("USER")``: exactly the same, except that when using a client-server topology, the
+  value will come from the environment of the client.
+- ``shell("get_login thisuser -t")``: execute the command, and use the output as the value. The
+  output will be trimmed of any trailing whitespace.
```

Comment:
and preceding whitespace?

Reply:
That is required to get it to belong to the bullet

Reply:
confusing comment - I meant:

```suggestion
  output will be trimmed of any trailing or preceding whitespace.
```

Reply:
OK, sorry, I'll put it in next time.